colossalai/fx/passes/algorithms/ckpt_solver_rotor.py

```diff
@@ -406,9 +417,18 @@ def solver_rotor(gm: ColoGraphModule,
 
     # found sequence
     sequence = _rec(chain, 0, chain.length, mem_slots - chain.cweight[0], opt_table)
-    _annotate_from_sequence(sequence, node_list)
+
+    # if solver failed, we don't need to annotate the graph
+    if not SOLVER_FAILED:
+        _annotate_from_sequence(sequence, node_list)
 
     # set __sequence__ attribute to GraphModule
-    setattr(gm, "__sequence__", sequence)
+    if SOLVER_FAILED:
+        setattr(gm, "__sequence__", None)
+    else:
+        setattr(gm, "__sequence__", sequence)
+
+    # set __opttable__ attribute to GraphModule
+    setattr(gm, "__opttable__", opt_table[0])
```

Comment:
why `opttable[0]`?

Reply:
because I think don't need `what` table to be attached to the graph module, the returned `opt_table[0]` by solver is a table containing `opt_table[0] = opt, opt_table[1] = what`, and I think we only need `opt` to check the value.

Reply:
I see